observer/observer_test.go

```diff
@@ -0,0 +1,141 @@
+package observer
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+	"time"
+
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	cmtjson "github.com/cometbft/cometbft/libs/json"
+	"github.com/cometbft/cometbft/libs/log"
+	coretypes "github.com/cometbft/cometbft/rpc/core/types"
+	"github.com/cometbft/cometbft/rpc/jsonrpc/types"
+	cmttypes "github.com/cometbft/cometbft/types"
+	proto "github.com/cosmos/gogoproto/proto"
+	"github.com/gorilla/websocket"
+	"github.com/stretchr/testify/require"
+
+	bridge "github.com/osmosis-labs/osmosis/v23/x/bridge/types"
+)
+
+const (
+	EventOutboundTransferType = "osmosis.bridge.v1beta1.EventOutboundTransfer"
+)
+
+var upgrader = websocket.Upgrader{}
+
+func readNewBlockEvent(path string) coretypes.ResultEvent {
+	dataStr, err := os.ReadFile(path)
+	if err != nil {
+		panic(err)
+	}
+	result := coretypes.ResultEvent{}
+	err = cmtjson.Unmarshal([]byte(dataStr), &result)
+	if err != nil {
+		panic(err)
+	}
+	return result
+}
+
+func readBlockResults(path string) coretypes.ResultBlockResults {
+	dataStr, err := os.ReadFile(path)
+	if err != nil {
+		panic(err)
+	}
+	result := coretypes.ResultBlockResults{}
+	err = json.Unmarshal([]byte(dataStr), &result)
+	if err != nil {
+		panic(err)
+	}
+	return result
+}
+
+func success(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "GET" {
```

Comment:
Go-way would be to use switch-case instead of if-else. So I suggest using it
```go
switch r.Method {
case http.MethodGet:
case http.MethodPost:
}
```